packages/react-reconciler/src/ReactFiberRoot.js

```diff
@@ -23,10 +23,16 @@ export type FiberRoot = {
   pendingChildren: any,
   // The currently active root fiber. This is the mutable root of the tree.
   current: Fiber,
+  // Whether this root is part of the root schedule. Different than whether it
+  // has remaining expiration time, because we don't remove roots from the
+  // schedule until the next time we search for work.
+  isScheduled: boolean,
```

Comment:
Let's do the back pointer instead so we can remove it immediately, or use a cyclic linked list where `null` means not part of the list.